iopipe/iopipe.py

```diff
@@ -1,217 +1,21 @@
-import concurrent.futures as futures
-import functools
-import inspect
-import logging
-import signal
-import warnings
+from .agent import IOpipeCore
 
-from .config import set_config
-from .context import ContextWrapper
-from .plugins import is_plugin
-from .report import Report
+from .contrib.eventinfo import EventInfoPlugin
+from .contrib.logger import LoggerPlugin
+from .contrib.profiler import ProfilerPlugin
+from .contrib.trace import TracePlugin
 
-logging.basicConfig()
 
-logger = logging.getLogger('iopipe')
-logger.setLevel(logging.INFO)
+class IOpipe(IOpipeCore):
+    """
+    The default IOpipe agent, pre-loaded with all the bundled plugins.
+    """
 
+    def __init__(self, *args, **kwargs):
+        configured_plugins = kwargs.pop('plugins', None)
+        plugins = [EventInfoPlugin(), LoggerPlugin(), ProfilerPlugin(), TracePlugin()]
+        if configured_plugins is not None and isinstance(plugins, list):
+            plugins = plugins + configured_plugins
```

Comment:
Huh, this de-dupes plugins?

Reply:
Yes, see `load_plugins` in `IOpipeCore`.